### PR TITLE
Make Prometheus build on Windows (without make).

### PR DESCRIPTION
**Pull request by smyrman**
Using `scripts/embed-static.sh` proved problematic on Windows, e.g. since "xxd" was not available or easy installable to me. Also, using make on Windows, proved to be somewhat problematic for me at least. So this pull requests does mainly two things:

1. Switched to using `go-bindata`(https://github.com/jteeuwen/go-bindata)  in place of `scripts/embed-static.sh`, and use go generate to create the binary blobs.
2. Support for building Prometheus without a `Makefile` (on Windows)

In addition, there seemed to be a minor typos in util/flock/flock_windows.go -- either that, or I am missing something...

The following build instructions should now work on both Linux and Windows, although only tested on Windows for now:

    cd "$GOPATH/src/github.com/prometheus/prometheus"
    go get -u ./...
    go generate ./...
    go build ./...
    go install ./...

Please note that this does not mean that Prometheus will now work on Windows.
I am able to start the server and access the WebUI, but there are still failing tests
within "github.com/prometheus/prometheus/storage/local".

### CODE REVIEW

**Comment by smyrman** [comment]
Seams this needs some more work. What is the best way for me to get your Makefile to download https://github.com/jteeuwen/go-bindata?

**Comment by fabxc** [comment]
This is great as we talked about switching to `go-bindata` just a few hours ago.

The best way to download would be `go get github.com/jteeuwen/go-bindata`. But we should just check in the generated go code as it is expected for `go-generate`. Makes things easier for 99% of people who just want to build from source.

**Comment by juliusv** [comment]
Thanks! This'll also need updates to https://github.com/prometheus/prometheus/blob/master/web/web.go#L101 when Prometheus is configured to use local assets during development.

**Comment by juliusv** [comment]
And this line too: https://github.com/prometheus/prometheus/blob/master/web/web.go#L139

File: web/Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ SUFFIXES:
 blob/files.go: $(shell find templates/ static/ -type f)
 	# Note that embed-static.sh excludes map files and the
```
Comment (juliusv) [inline web/Makefile]: It seems like this comment can now simply be deleted.

File: web/Makefile [hunk 2]
```diff
@@ -20,7 +20,7 @@ SUFFIXES:
 blob/files.go: $(shell find templates/ static/ -type f)
```
Comment (juliusv) [inline web/Makefile]: The paths in the dependency here need to be updated too.

File: web/Makefile [hunk 3]
```diff
@@ -20,7 +20,7 @@ SUFFIXES:
...
 	# Note that embed-static.sh excludes map files and the
 	# non-minified bootstrap files.
-	../scripts/embed-static.sh static templates | $(GOFMT) > $@
+	go generate ./blob
 
 clean:
```
Comment (juliusv) [inline web/Makefile]: I agree with @fabxc that `files.go` should be checked in now, so how about just getting rid of the `clean` target here and also removing the reference to it in https://github.com/prometheus/prometheus/blob/master/Makefile#L58.

File: web/Makefile [hunk 4]
```diff
@@ -20,7 +20,7 @@ SUFFIXES:
 blob/files.go: $(shell find templates/ static/ -type f)
 	# Note that embed-static.sh excludes map files and the
 	# non-minified bootstrap files.
-	../scripts/embed-static.sh static templates | $(GOFMT) > $@
+	go generate ./blob
```
Comment (juliusv) [inline web/Makefile]: Let's use `$(GO)` instead of `go` here to use the same Go environment as in the main Makefile.

**Comment by smyrman** [comment]
Tested to build with/without Makefile on Linux. Go unit-tests also run with no failures. go-bindata should now be downloaded automatically when building using the Makefiles.

**Comment by smyrman** [comment]
As far as I can see, the Prometheus admin UI seems to be working fine as well.

Let me know if you prefer me to squash the commits, or if there is any other fixups you want me to do:-)

**Comment by juliusv** [comment]
Nice! Everything works for me and looks good (well, on Linux, don't have a Windows around to test). Indeed, could you squash the commits?

**Comment by smyrman** [comment]
@juliusv, it's done.

**Comment by juliusv** [comment]
Woot, thanks!

**Comment by jimmidyson** [comment]
So does this mean Prometheus runs on Windows? I'm asking for a friend :stuck_out_tongue_winking_eye:

**Comment by fabxc** [comment]
None of us has tried it personally, but I assume it does.

**Comment by jimmidyson** [comment]
@smyrman Do you run Prometheus on Windows?

**Pull request merged**